File: angular-app/src/app/_model/Comment.ts
import { Offer } from "./Offer";
import { User } from "./User";

export class Comment {
    id: number | null;
    content: string;
    user: User;
    offer: Offer;

    constructor(id: number | null, content: string, user: User, offer: Offer) {
        this.id = id;
        this.content = content;
        this.user = user;
        this.offer = offer;
    }
}

File: SpringBootApp/src/main/java/net/etfbl/webshop/models/entities/Comment.java
package net.etfbl.webshop.models.entities;

import jakarta.persistence.*;
import lombok.Data;
import net.etfbl.webshop.base.BaseEntity;

@Data
@Entity
@Table(name = "comment")
public class Comment implements BaseEntity<Integer> {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Basic
    @Column(name = "content", nullable = false, length = -1)
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_user", referencedColumnName = "id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_offer", referencedColumnName = "id", nullable = false)
    private Offer offer;

}
